src/utils/http.tsx

```diff
@@ -7,30 +7,34 @@ type Config = {
 
 const request = (url: string, method: string, body: Object | null, config: Config) => {
     const token = localStorage['token'];
-    let headers = (!config.noAuth && token) ? {"Content-Type": "application/json", Authorization: "Bearer " + token} : {"Content-Type": "application/json"};
+    let headers = (!config.noAuth && token) ? {
+        "Content-Type": "application/json",
+        Authorization: "Bearer " + token
+    } : {"Content-Type": "application/json"};
     const configuration: Object = {
         method: method,
         body: body ? JSON.stringify(body) : undefined,
         headers: headers,
     };
     return fetch(baseUrl + url, configuration)
```

Comment:
Some comments here:
- I don't see the advantage of creating the checkStatus and the ParseJSON (no so important)
- You should use the res.ok to check if the request was successful or not.
- Is better if you throw the error as a string instead of a promise, this will avoid making the error.then() that I saw around the code.

The code should be something like this:
```
.then(response => {
            if(response.ok) return response.json();
            // if the token has expired log out the user
            else if(response.status === 403 && localStorage['token']) {
                localStorage.removeItem('token');
                localStorage.removeItem('admin');
                window.location.reload();
            }
            // if an error occurs on the server return the error as text (because we are returning plain text errors)
            return response.text().then(error => {
                throw (error)
            })
        })
        // Catch connection errors and the error throw above.
        .catch(error => {
            throw(error)
        })
```